### ISSUE TITLE
EM_JS doesn't insert semicolons

### ISSUE THREAD

**Issue by curiousdannii** [emscripten-core/emscripten#11793]
The EM_JS macro requires explicit semicolons, if you write code in the now common semicolon-less style it will complain.

Adding semicolons is likely to be tricky, but maybe it would be possible to warn if it doesn't see them?

Or could it not output the function with all linebreaks removed, so that the compressor can handle the semicolon insertion?

**Comment by kripken** [comment]
Yes, if the problem is automatic semicolon insertion then if we change whitespace that could be the issue perhaps. Does the problem only happen in minified builds (-O2 and above)? Can look at the EMCC_DEBUG output files to see at which stage the problem happens.

**Comment by curiousdannii** [comment]
Can double check tomorrow, but I'm pretty sure the problem does happen with -O0, but not with --profiler.

**Comment by curiousdannii** [comment]
This should still be fixed
